internal/grid/types.go

```diff
@@ -121,6 +140,20 @@ func NewMSSWith(m map[string]string) *MSS {
 	return &m2
 }
 
+var mssPool = sync.Pool{
+	New: func() interface{} {
+		return make(map[string]string, 5)
+	},
+}
+
+// Recycle the underlying map.
+func (m *MSS) Recycle() {
+	if m != nil && *m != nil {
+		mssPool.Put(map[string]string(*m))
```

Comment:
Alternatively, clear the contents before we insert back into the pool ?
(so we know there never is any left over content).